From e48a832aadd0cef5e469801c7bf844787b361796 Mon Sep 17 00:00:00 2001
From: Matthew Schwartz <matthew.schwartz@linux.dev>
Date: Sat, 20 Dec 2025 19:59:53 -0800
Subject: [PATCH 050/176] [FROM-ML] misc: rtsx_pcr: prevent pm_schedule_suspend
 for RTS525A

The chip's pm_schedule_suspend sequence can sometimes cause SD card
reinitialization failures with "__mmc_poll_for_busy" after reaching the
end of its operations.

As a workaround, skip pm_schedule_suspend for RTS525A card readers.

Signed-off-by: Matthew Schwartz <matthew.schwartz@linux.dev>
---
 drivers/misc/cardreader/rtsx_pcr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/misc/cardreader/rtsx_pcr.c b/drivers/misc/cardreader/rtsx_pcr.c
index c4d54ca2fa80..ff41842b4dd9 100644
--- a/drivers/misc/cardreader/rtsx_pcr.c
+++ b/drivers/misc/cardreader/rtsx_pcr.c
@@ -1760,7 +1760,7 @@ static int rtsx_pci_runtime_idle(struct device *device)
 
 	mutex_unlock(&pcr->pcr_mutex);
 
-	if (pcr->rtd3_en)
+	if (pcr->rtd3_en && PCI_PID(pcr) != PID_525A)
 		pm_schedule_suspend(device, 10000);
 
 	return -EBUSY;
-- 
2.55.0

